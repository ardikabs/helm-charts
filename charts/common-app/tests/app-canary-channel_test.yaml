suite: app canary channel test
templates:
  - "deployment.yaml"
  - "service.yaml"
  - "ingress.yaml"
  - "hpa.yaml"
  - "pdb.yaml"
values:
  - "values/full_app-canary-channel.yaml"
release:
  name: podinfo-canary
  namespace: testing
tests:
  - it: should create Deployment
    template: deployment.yaml
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -canary$
      - equal:
          path: spec.template.spec.containers[?(@.name == "main")].image
          value: ghcr.io/ardikabs/etc/podinfo:6.3.3
      - matchSnapshot:
          path: spec
  - it: should create a Service
    template: service.yaml
    asserts:
      - isKind:
          of: Service
      - matchRegex:
          path: metadata.name
          pattern: -canary$
      - hasDocuments:
          count: 1
      - equal:
          path: spec.selector
          value:
            app.kubernetes.io/name: podinfo
            app.kubernetes.io/channel: canary
      - matchSnapshot:
          path: spec
  - it: should not create an Ingress
    template: ingress.yaml
    asserts:
      - hasDocuments:
          count: 0
  - it: should not create a HPA
    template: hpa.yaml
    asserts:
      - hasDocuments:
          count: 0
  - it: should not create a PDB
    template: pdb.yaml
    asserts:
      - hasDocuments:
          count: 0
