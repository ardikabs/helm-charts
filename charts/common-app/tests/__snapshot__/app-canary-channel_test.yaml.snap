should create Deployment:
  1: |
    minReadySeconds: 0
    selector:
      matchLabels:
        app.kubernetes.io/channel: canary
        app.kubernetes.io/name: podinfo
    strategy:
      rollingUpdate:
        maxSurge: 30%
        maxUnavailable: 0
      type: RollingUpdate
    template:
      metadata:
        annotations:
          argocd.argoproj.io/sync-options: PruneLast=true
          prometheus.io/port: "8888"
          prometheus.io/scrape: "true"
        labels:
          app.kubernetes.io/channel: canary
          app.kubernetes.io/domain: common
          app.kubernetes.io/instance: podinfo-canary
          app.kubernetes.io/managed-by: manual
          app.kubernetes.io/name: podinfo
          app.kubernetes.io/team: core
          app.kubernetes.io/version: 6.3.3
          app.kubernetes.io/workload-type: java
          helm.sh/chart: common-app-0.4.0
      spec:
        affinity:
          nodeAffinity:
            preferredDuringSchedulingIgnoredDuringExecution:
              - preference:
                  matchExpressions:
                    - key: label-1
                      operator: In
                      values:
                        - key-1
                weight: 1
            requiredDuringSchedulingIgnoredDuringExecution:
              nodeSelectorTerms:
                - matchExpressions:
                    - key: kubernetes.io/os
                      operator: In
                      values:
                        - linux
          podAffinity:
            preferredDuringSchedulingIgnoredDuringExecution:
              - podAffinityTerm:
                  labelSelector:
                    matchExpressions:
                      - key: app
                        operator: In
                        values:
                          - helloworld
                  topologyKey: kubernetes.io/hostname
                weight: 100
          podAntiAffinity: {}
        containers:
          - args:
              - --loglevel warn
              - --logoutput stdout
            command:
              - apiserver
            env:
              - name: HTTP_PORT
                value: "8080"
            envFrom:
              - secretRef:
                  name: another-secret
              - configMapRef:
                  name: another-configmap
            image: ghcr.io/ardikabs/etc/podinfo:6.3.3
            imagePullPolicy: IfNotPresent
            livenessProbe:
              exec:
                command:
                  - podcli
                  - check
                  - http
                  - localhost:9898/readyz
              failureThreshold: 3
              initialDelaySeconds: 10
              periodSeconds: 10
              successThreshold: 1
              timeoutSeconds: 1
            name: main
            ports:
              - containerPort: 9898
                name: http
                protocol: TCP
            readinessProbe:
              exec:
                command:
                  - podcli
                  - check
                  - http
                  - localhost:9898/readyz
              failureThreshold: 3
              initialDelaySeconds: 10
              periodSeconds: 10
              successThreshold: 1
              timeoutSeconds: 1
            resources:
              limits:
                cpu: 100m
                memory: 90Mi
              requests:
                cpu: 100m
                memory: 90Mi
            volumeMounts:
              - mountPath: /usr/share/app-config-file
                name: app-config-file
              - mountPath: /usr/share/app-secret-file
                name: app-secret-file
        imagePullSecrets:
          - name: github-cred
        nodeSelector:
          webserver: public
        securityContext:
          fsGroup: 2000
          runAsGroup: 3000
          runAsUser: 1000
        serviceAccountName: default
        terminationGracePeriodSeconds: 30
        tolerations:
          - operator: Exists
        volumes:
          - emptyDir: {}
            name: shared-volume
          - configMap:
              items:
                key: application.properties
                mode: 420
                path: application.properties
              name: podinfo-canary
            name: app-config-file
          - name: app-secret-file
            secret:
              items:
                key: secret.properties
                mode: 420
                path: secret.properties
              secretName: podinfo-canary
should create a Service:
  1: |
    ports:
      - name: http
        port: 80
        protocol: TCP
        targetPort: http
    selector:
      app.kubernetes.io/channel: canary
      app.kubernetes.io/name: podinfo
    type: ClusterIP
